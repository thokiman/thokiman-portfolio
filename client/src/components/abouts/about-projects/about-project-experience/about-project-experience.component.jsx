import React from "react";

import {
  AboutDescriptionExperienceProject,
  AboutExperienceProjectAtttributes,
  AboutExperienceProjectContainer,
  AboutExperienceProjectTitle,
  AboutNameExperienceProject,
} from "./about-project-experience.styles";

export const AboutExperienceProject = ({ experience }) => {
  return (
    <AboutExperienceProjectContainer>
      <AboutExperienceProjectTitle>
        Experience Project
      </AboutExperienceProjectTitle>
      {experience.map(({ id, name, description }) => {
        return (
          <AboutExperienceProjectAtttributes key={id}>
            <AboutNameExperienceProject>{name}</AboutNameExperienceProject>
            <AboutDescriptionExperienceProject>
              {description}
            </AboutDescriptionExperienceProject>
          </AboutExperienceProjectAtttributes>
        );
      })}
    </AboutExperienceProjectContainer>
  );
};

export default AboutExperienceProject;
