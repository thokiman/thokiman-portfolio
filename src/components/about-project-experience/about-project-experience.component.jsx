import React from "react";

import "./about-project-experience.styles.scss";

const AboutExperienceProject = ({ experience }) => {
  return (
    <div className="experience-project-container">
      <div className="experience-project-title">Experience Project</div>
      {experience.map(({ id, name, description }) => {
        return (
          <div key={id} className="experience-project">
            <div className="name-experience-project">{name}</div>
            <div className="description-experience-project">{description}</div>
          </div>
        );
      })}
    </div>
  );
};

export default AboutExperienceProject;
