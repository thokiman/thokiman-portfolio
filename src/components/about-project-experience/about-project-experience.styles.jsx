import styled, { css } from "styled-components";

export const AboutExperienceProjectContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 95%;
  margin: 0 auto;
`;

export const AboutExperienceProjectTitle = styled.div`
  font-size: 400%;
  padding: 10px;
  color: #e3e1e4;
  letter-spacing: 0.04em;
  text-align: left;
  display: flex;
  align-items: center;
`;

export const AboutExperienceProjectAtttributes = styled.div`
  width: 100%;
  height: 100%;
  display: grid;
  grid-auto-columns: repeat(4, 1fr);
  grid-auto-rows: repeat(3, 1fr);
  gap: 2px 2px;
  grid-template-areas:
    "name-experience-project name-experience-project name-experience-project name-experience-project"
    "description-experience-project description-experience-project description-experience-project description-experience-project"
    "description-experience-project description-experience-project description-experience-project description-experience-project";
`;
const AboutExperienceProjectStyles = css`
  display: flex;
  align-items: center;
  color: #e3e1e4;
  text-align: left;
`;
export const AboutNameExperienceProject = styled.div`
  ${AboutExperienceProjectStyles}
  grid-area: name-experience-project;
  width: 99%;
  font-size: 200%;
  padding: 10px;
  letter-spacing: 0.09em;
`;

export const AboutDescriptionExperienceProject = styled.div`
  ${AboutExperienceProjectStyles}
  border-bottom: #74c6d4 solid 1px;
  width: 100%;
  grid-area: description-experience-project;
  font-size: 130%;
  padding: 10px 0;
  line-height: 1.1em;
  letter-spacing: 0.05em;
  margin: 10px auto;
`;
AboutExperienceProjectContainer.displayName = "AboutExperienceProjectContainer";
AboutExperienceProjectTitle.displayName = "AboutExperienceProjectTitle";
AboutExperienceProjectAtttributes.displayName =
  "AboutExperienceProjectAtttributes";
AboutNameExperienceProject.displayName = "AboutNameExperienceProject";
AboutDescriptionExperienceProject.displayName =
  "AboutDescriptionExperienceProject";
