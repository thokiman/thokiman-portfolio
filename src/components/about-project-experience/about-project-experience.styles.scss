.experience-project-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 95%;
    margin: 0 auto;

    .experience-project-title {
        font-size: 400%;
        padding: 10px;
        color: #e3e1e4;
        letter-spacing: 0.04em;
        text-align: left;
        display: flex;
        align-items: center;
    }

    .experience-project {

        width: 100%;
        height: 100%;
        display: grid;
        grid-auto-columns: repeat(4, 1fr);
        grid-auto-rows: repeat(3, 1fr);
        gap: 2px 2px;
        grid-template-areas:
            "name-experience-project name-experience-project name-experience-project name-experience-project"
            "description-experience-project description-experience-project description-experience-project description-experience-project"
            "description-experience-project description-experience-project description-experience-project description-experience-project";
    }







    .name-experience-project {

        width: 99%;

        grid-area: name-experience-project;
        font-size: 200%;
        padding: 10px;
        color: #e3e1e4;
        letter-spacing: 0.09em;
        text-align: left;
        display: flex;
        align-items: center;
    }



    .description-experience-project {
        border-bottom: #74c6d4 solid 1px;
        width: 100%;

        grid-area: description-experience-project;
        font-size: 150%;
        padding: 10px 0;
        display: flex;
        align-items: center;
        text-align: left;
        line-height: 1.1em;
        color: #e3e1e4;
        letter-spacing: 0.05em;
        margin: 10px auto;

    }
}