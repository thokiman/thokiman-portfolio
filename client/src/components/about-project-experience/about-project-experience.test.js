import React from "react";
import { shallow } from "enzyme";

import { AboutExperienceProject } from "./about-project-experience.component";

describe("AboutExperienceProject component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      experience: mockArray,
    };
    wrapper = shallow(<AboutExperienceProject {...mockProps} />);
  });

  it("should render AboutExperienceProject  component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
