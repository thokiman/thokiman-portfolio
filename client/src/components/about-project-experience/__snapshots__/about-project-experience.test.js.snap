// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutExperienceProject component should render AboutExperienceProject  component 1`] = `ShallowWrapper {}`;
